magprime/automated_emails.py

```diff
@@ -15,7 +15,7 @@ def __init__(self, event):
     SeasonSupporterEmail(_event)
 
 
-AutomatedEmail(Attendee, 'MAGFest schedule, maps, and other FAQs', 'precon_faqs.html', lambda a: days_before(7, c.EPOCH))
+AutomatedEmail(Attendee, 'MAGFest schedule, maps, and other FAQs', 'precon_faqs.html', lambda a: days_before(7, c.EPOCH) and ((a.badge_status != (c.INVALID_STATUS or c.DEFERRED_STATUS)) or (a.paid != NOT_PAID)))
```

Comment:
```
a.badge_status != (c.INVALID_STATUS or c.DEFERRED_STATUS)
```

needs to say

```
a.badge_status not in [c.INVALID_STATUS, c.DEFERRED_STATUS]
```

This isn't just a style issue; consider the following:

```
>>> c.INVALID_STATUS
229301191
>>> c.DEFERRED_STATUS
86547910
>>> c.INVALID_STATUS or c.DEFERRED_STATUS
229301191
```

So what's happening there?  When you say ``a or b`` then the value of that expression will result in the value of ``a`` if a is truthy, otherwise the expression will result in the value of ``b``.  In this case we want to make sure that the badge has neither status.

Echoing Vicki's comment, you need to replace the ``or`` with an ``and``.  You can also get rid of all of the parenthesis around your conditional clauses.